File: Backend/Entities/TaskItem.cs
namespace Backend.Entities;

public class TaskItem
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string Status { get; set; } = "Todo";
    public string Priority { get; set; } = "Medium";
    public DateTime? DueDate { get; set; }
    public int BoardId { get; set; }
    public Board? Board { get; set; }
}

File: Frontend/src/features/tasks/types/TaskItem.ts
export type TaskStatus = "Todo" | "InProgress" | "Done";
export type TaskPriority = "Low" | "Medium" | "High";

export interface TaskItem {
    id: number;
    title: string;
    description?: string;
    status: TaskStatus;
    boardId: number;
    priority: TaskPriority;
    dueDate?: string;
}
